File: 백준/Silver/30804. 과일 탕후루/과일 탕후루.py
from sys import stdin
from itertools import combinations


input = stdin.readline

n = int(input())
# 과일 종류는 1 ~ 9 숫자로 표시
fruits = list(map(int, input().rsplit()))

# 9C2 = 9 * 8 / 2
answer = 0
for case in combinations(range(1, 10), 2):
    cnt = 0
    for fruit in fruits:
        if fruit in case:
            cnt += 1
        else:
            answer = max(answer, cnt)
            cnt = 0
            
    answer = max(answer, cnt)
    
print(answer)
        

File: 백준/Silver/30804. 과일 탕후루/과일 탕후루.java
import java.io.*;
import java.util.*;

public class Main {
    static int n;
    static int[] fruits;
    static BufferedReader br;
    static StringTokenizer st;

    public static void main(String[] args) throws Exception {
        br = new BufferedReader(new InputStreamReader(System.in));

        n = Integer.parseInt(br.readLine());

        fruits = new int[n];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            fruits[i] = Integer.parseInt(st.nextToken());
        }

        int answer = 0;
        for (int i = 1; i < 10; i++) {
            for (int j = i + 1; j < 10; j++) {
                int cnt = 0;
                for (int fruit : fruits) {
                    if (fruit == i || fruit == j) {
                        cnt++;
                    } else {
                        answer = Math.max(answer, cnt);
                        cnt = 0;
                    }
                }
                answer = Math.max(answer, cnt);
            }
        }
        System.out.println(answer);
    }
}

